### ISSUE TITLE
Would it be possible to upgrade newer minimum required cmake version ? like 3.0 or 3.5

### ISSUE THREAD

**Issue by 9prady9** [CNugteren/CLBlast#407]
I am not sure if VERSION variables aren't set with older versions prior to 3.0, but they are definitely set with CMake versions above 3.0. It would be nice to use that feature to avoid the warnings in downstream projects when used via `add_subdirectory` approach.

Also, quite a few CMake variables don't have project name prefix causing name conflicts and/or confusion with downstream project's cache options when CLBlast is used via `add_subdirectory` command.

Should you guys be interested in the above, I can send in a PR soon with those changes.

Note: Ubuntu 16.04 uses cmake 3.5.1, although I believe CentOS 6 is still is at 2.8.12 - I am not 100% sure of centos 6.

**Comment by CNugteren** [comment]
About the second part, renaming some CMake variables, yes, I would be happy to accept such a PR.

About the first part, I'm not sure what you mean exactly, and therefore also don't know if it is worth bumping the minimum required CMake version. Do you have a link to some documentation or an example of this perhaps? Is it [this](https://cmake.org/cmake/help/v3.13/variable/CMAKE_PROJECT_VERSION.html) you are talking about?

**Comment by 9prady9** [comment]
> About the second part, renaming some CMake variables, yes, I would be happy to accept such a PR.

Sure, will send soon.

> Is it this you are talking about?

That and some new policies that are throwing warnings like the following I include CLBlast via `add_subdirectory` approach into a downstream project:
- https://cmake.org/cmake/help/v3.0/policy/CMP0048.html
- https://cmake.org/cmake/help/v3.3/policy/CMP0063.html
- https://cmake.org/cmake/help/git-stage/policy/CMP0077.html

**Comment by CNugteren** [comment]
OK thanks for the details. Now that would then indeed change the version requirement. Since this all just happens at the start (right?) of the code, isn't it perhaps possible to do something along the lines of this:
```
cmake_minimum_required(VERSION 2.8.11)
if(${CMAKE_VERSION} VERSION_LESS "3.3.0")   # or whatever is the minimum required for these new features 
    project("clblast" C CXX)
else()
    project("clblast" C CXX VERSION 1.5.1)
endif()
```
That will make it a bit less readable but at least you can use the new feature for your purpose and others without a new version of CMake can still easily build CLBlast.

**Comment by 9prady9** [comment]
Yes, something like that should work fine. Set policies are ignored I believe if they are not available in current version. So that should be also good.
